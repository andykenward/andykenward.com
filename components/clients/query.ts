import { DocumentNode } from "graphql"
import gql from "graphql-tag"

export const FRAGMENT_CLIENT: DocumentNode = gql`
  fragment FragmentClient on Client {
    id
    title
    href
  }
`
