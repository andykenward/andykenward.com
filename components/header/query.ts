import { DocumentNode, FragmentSpreadNode } from "graphql"
import gql from "graphql-tag"

import { FRAGMENT_LINK } from "../link/query"

export const FRAGMENT_ME: FragmentSpreadNode = gql`
  fragment FragmentMe on Me {
    id
    name
    role
    update
    links {
      ...FragmentLink
    }
  }
  ${FRAGMENT_LINK}
`

export const QUERY_ME: DocumentNode = gql`
  query Me {
    me(where: { id: "cjzkx8reiebu10941932j2b3x" }) {
      ...FragmentMe
    }
  }
  ${FRAGMENT_ME}
`
