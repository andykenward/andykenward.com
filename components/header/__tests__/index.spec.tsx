import { useQuery } from "@apollo/react-hooks"
import { wait } from "@apollo/react-testing"
import { act, RenderResult } from "@testing-library/react"

import { Header } from ".."
import { getMock, getMocksError, renderWithApp } from "../../../lib/test-utils"
import { QUERY_ME } from "../query"
import {
  HEADER_RESULT_DATA,
  HEADER_RESULT_DATA_NO_LINKS,
  HEADER_RESULT_DATA_NO_NAME,
  HEADER_RESULT_DATA_NO_ROLE,
  HEADER_RESULT_DATA_NO_UPDATE,
} from "./__fixtures__"

describe("<Header />", () => {
  it("should handle Me query", async () => {
    const MOCKS = getMock(QUERY_ME, HEADER_RESULT_DATA)
    const Component = () => {
      const { data, loading } = useQuery(QUERY_ME)
      if (!loading) {
        expect(data).toEqual(HEADER_RESULT_DATA)
      }
      return null
    }
    act(() => {
      renderWithApp(<Component />, MOCKS)
    })
  })

  it("renders null on loading", async () => {
    const MOCKS = getMock(QUERY_ME, HEADER_RESULT_DATA)

    await act(async () => {
      const { container } = renderWithApp(<Header />, MOCKS)

      expect(container.firstChild).toBeNull()
      expect(container.firstChild).toMatchSnapshot()
    })
  })

  it("renders null on error", async () => {
    const MOCKS = getMocksError(QUERY_ME)

    await act(async () => {
      const { container } = renderWithApp(<Header />, MOCKS)

      await wait(0)

      expect(container.firstChild).toBeNull()
      expect(container.firstChild).toMatchSnapshot()
    })
  })

  it("renders null on undefined data", async () => {
    const MOCKS = getMock(QUERY_ME)
    await act(async () => {
      const { container } = renderWithApp(<Header />, MOCKS)

      await wait(0)

      expect(container.firstChild).toBeNull()
      expect(container.firstChild).toMatchSnapshot()
    })
  })

  const TEST_ID_NAME = "name"
  const TEST_ID_ROLE = "role"
  const TEST_ID_UPDATE = "update"
  const TEST_ID_LINKS = "links"
  const TEST_ID_LINK = "link"

  it("should render correctly", async () => {
    let root: RenderResult
    const MOCKS = getMock(QUERY_ME, HEADER_RESULT_DATA)
    await act(async () => {
      root = renderWithApp(<Header />, MOCKS)
    })
    await act(async () => {
      await wait(0)
    })
    act(() => {
      const { container, getByTestId, getAllByTestId } = root
      expect(getByTestId(TEST_ID_NAME).tagName).toEqual("H1")
      expect(getByTestId(TEST_ID_NAME)).toHaveTextContent("Andy Kenward")
      expect(getByTestId(TEST_ID_ROLE).tagName).toEqual("H2")
      expect(getByTestId(TEST_ID_ROLE)).toHaveTextContent(
        "Senior Front-End Engineer"
      )
      expect(getByTestId(TEST_ID_UPDATE).tagName).toEqual("H3")
      expect(getByTestId(TEST_ID_UPDATE)).toHaveTextContent("Hire Me")
      const links = getByTestId(TEST_ID_LINKS)
      expect(links).not.toBeNull()
      expect(getAllByTestId(TEST_ID_LINK).length).toEqual(1)
      expect(getByTestId(TEST_ID_LINK).getAttribute("href")).toEqual("/link")
      expect(getByTestId(TEST_ID_LINK)).toHaveTextContent("Github")

      expect(container.firstChild).toMatchSnapshot()
    })
  })

  it("should render when `name` is `null`", async () => {
    const MOCKS = getMock(QUERY_ME, HEADER_RESULT_DATA_NO_NAME)
    let root: RenderResult
    await act(async () => {
      root = renderWithApp(<Header />, MOCKS)
    })
    await act(async () => {
      await wait(0)
    })
    act(() => {
      const { container, queryByTestId, getByTestId, getAllByTestId } = root
      expect(queryByTestId(TEST_ID_NAME)).toBeNull()
      expect(getByTestId(TEST_ID_ROLE)).toHaveTextContent(
        "Senior Front-End Engineer"
      )
      expect(getByTestId(TEST_ID_UPDATE)).toHaveTextContent("Hire Me")
      expect(getByTestId(TEST_ID_LINKS)).not.toBeNull()
      expect(getAllByTestId(TEST_ID_LINK).length).toEqual(1)

      expect(container.firstChild).toMatchSnapshot()
    })
  })

  it("should render when `role` is `null`", async () => {
    const MOCKS = getMock(QUERY_ME, HEADER_RESULT_DATA_NO_ROLE)
    let root: RenderResult
    await act(async () => {
      root = renderWithApp(<Header />, MOCKS)
    })
    await act(async () => {
      await wait(0)
    })
    act(() => {
      const { container, queryByTestId, getByTestId, getAllByTestId } = root
      expect(getByTestId(TEST_ID_NAME)).toHaveTextContent("Andy Kenward")
      expect(queryByTestId(TEST_ID_ROLE)).toBeNull()
      expect(getByTestId(TEST_ID_UPDATE)).toHaveTextContent("Hire Me")
      expect(getByTestId(TEST_ID_LINKS)).not.toBeNull()
      expect(getAllByTestId(TEST_ID_LINK).length).toEqual(1)

      expect(container.firstChild).toMatchSnapshot()
    })
  })

  it("should render when `update` is `null`", async () => {
    const MOCKS = getMock(QUERY_ME, HEADER_RESULT_DATA_NO_UPDATE)
    let root: RenderResult
    await act(async () => {
      root = renderWithApp(<Header />, MOCKS)
    })
    await act(async () => {
      await wait(0)
    })
    act(() => {
      const { container, queryByTestId, getByTestId, getAllByTestId } = root
      expect(queryByTestId(TEST_ID_UPDATE)).toBeNull()
      expect(getByTestId(TEST_ID_NAME)).toHaveTextContent("Andy Kenward")
      expect(getByTestId(TEST_ID_ROLE)).toHaveTextContent(
        "Senior Front-End Engineer"
      )
      expect(getByTestId(TEST_ID_LINKS)).not.toBeNull()
      expect(getAllByTestId(TEST_ID_LINK).length).toEqual(1)

      expect(container.firstChild).toMatchSnapshot()
    })
  })

  it("should render when `links` is `null`", async () => {
    const MOCKS = getMock(QUERY_ME, HEADER_RESULT_DATA_NO_LINKS)
    let root: RenderResult
    await act(async () => {
      root = renderWithApp(<Header />, MOCKS)
    })
    await act(async () => {
      await wait(0)
    })
    act(() => {
      const { container, queryByTestId, getByTestId } = root
      expect(getByTestId(TEST_ID_NAME)).toHaveTextContent("Andy Kenward")
      expect(getByTestId(TEST_ID_ROLE)).toHaveTextContent(
        "Senior Front-End Engineer"
      )
      expect(getByTestId(TEST_ID_UPDATE)).toHaveTextContent("Hire Me")
      expect(queryByTestId(TEST_ID_LINKS)).toBeNull()
      expect(queryByTestId(TEST_ID_LINK)).toBeNull()
      expect(container.firstChild).toMatchSnapshot()
    })
  })
})
