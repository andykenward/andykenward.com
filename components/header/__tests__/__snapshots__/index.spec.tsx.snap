// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders null on error 1`] = `null`;

exports[`<Header /> renders null on loading 1`] = `null`;

exports[`<Header /> renders null on undefined data 1`] = `null`;

exports[`<Header /> should render correctly 1`] = `
.emotion-12 {
  background-color: rgb(216,234,204);
  display: grid;
  grid-template-columns: 32px [center-start] 1fr [center-end] 32px;
  padding-top: 4rem;
  padding-bottom: 4rem;
}

@media screen and (min-width:40em) {
  .emotion-12 {
    grid-template-columns: minmax(32px,15%) [center-start] 1fr [center-end] minmax(32px,15%);
  }
}

@media screen and (min-width:40em) {
  .emotion-12 {
    padding-top: 8rem;
    padding-bottom: 8rem;
  }
}

.emotion-0 {
  font-family: "Cooper Light",Georgia,Times,"Times New Roman",serif;
  font-weight: 300;
  font-size: 3rem;
  line-height: 1.090909091;
  text-transform: capitalize;
  color: rgb(36,28,21);
  margin: 0;
  grid-column: center-start/center-end;
  word-break: break-word;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    font-size: 4rem;
  }
}

@media screen and (min-width:52em) {
  .emotion-0 {
    font-size: 5rem;
  }
}

.emotion-2 {
  font-family: Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif;
  font-weight: 400;
  font-size: 2.25rem;
  -webkit-letter-spacing: -0.03125rem;
  -moz-letter-spacing: -0.03125rem;
  -ms-letter-spacing: -0.03125rem;
  letter-spacing: -0.03125rem;
  line-height: 1.25;
  text-transform: capitalize;
  color: rgb(36,28,21);
  margin: 0;
  margin-bottom: 0.5rem;
  grid-column: center-start/center-end;
  word-break: break-word;
}

@media screen and (min-width:40em) {
  .emotion-2 {
    font-size: 2.875rem;
  }
}

.emotion-4 {
  font-family: Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif;
  font-weight: 400;
  font-size: 1.25rem;
  -webkit-letter-spacing: -0.0125rem;
  -moz-letter-spacing: -0.0125rem;
  -ms-letter-spacing: -0.0125rem;
  letter-spacing: -0.0125rem;
  line-height: 1.25;
  text-transform: capitalize;
  color: rgb(36,28,21);
  margin: 0;
  margin-bottom: 1rem;
  grid-column: center-start/center-end;
  word-break: break-word;
}

@media screen and (min-width:40em) {
  .emotion-4 {
    font-size: 1.75rem;
  }
}

.emotion-10 {
  grid-column: center-start/center-end;
}

.emotion-8 {
  font-family: Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  color: rgb(36,28,21);
  margin: 0;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  margin-right: 0.5rem;
  margin-bottom: 0.5rem;
  word-break: break-word;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-transition: all 0.15s linear;
  transition: all 0.15s linear;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  box-shadow: inset 0 0 0 0.0625rem rgba(36,28,21,0.4);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: transparent;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: middle;
}

.emotion-8:visited:hover,
.emotion-8:hover,
.emotion-8:focus {
  color: inherit;
}

.emotion-8:visited {
  color: inherit;
}

.emotion-8:hover,
.emotion-8:focus {
  color: rgb(64,59,59);
  box-shadow: inset 0 0 0 0.0625rem rgb(216,234,204);
  background-color: rgb(216,234,204);
}

<header
  class="emotion-12 emotion-11"
>
  <h1
    class="emotion-0 emotion-1"
    data-testid="name"
  >
    Andy Kenward
  </h1>
  <h2
    class="emotion-2 emotion-1"
    data-testid="role"
  >
    Senior Front-End Engineer
  </h2>
  <h3
    class="emotion-4 emotion-1"
    data-testid="update"
  >
    Hire Me
  </h3>
  <h4
    class="emotion-4 emotion-1"
    data-testid="available"
  >
    Available from 
    April 2020
  </h4>
  <div
    class="emotion-10 emotion-11"
    data-testid="links"
  >
    <a
      class="emotion-8 emotion-1"
      data-testid="link"
      href="/link"
      rel="noopener noreferrer"
      target="_blank"
    >
      Github
    </a>
  </div>
</header>
`;

exports[`<Header /> should render when \`available\` is \`null\` 1`] = `
.emotion-10 {
  background-color: rgb(216,234,204);
  display: grid;
  grid-template-columns: 32px [center-start] 1fr [center-end] 32px;
  padding-top: 4rem;
  padding-bottom: 4rem;
}

@media screen and (min-width:40em) {
  .emotion-10 {
    grid-template-columns: minmax(32px,15%) [center-start] 1fr [center-end] minmax(32px,15%);
  }
}

@media screen and (min-width:40em) {
  .emotion-10 {
    padding-top: 8rem;
    padding-bottom: 8rem;
  }
}

.emotion-0 {
  font-family: "Cooper Light",Georgia,Times,"Times New Roman",serif;
  font-weight: 300;
  font-size: 3rem;
  line-height: 1.090909091;
  text-transform: capitalize;
  color: rgb(36,28,21);
  margin: 0;
  grid-column: center-start/center-end;
  word-break: break-word;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    font-size: 4rem;
  }
}

@media screen and (min-width:52em) {
  .emotion-0 {
    font-size: 5rem;
  }
}

.emotion-2 {
  font-family: Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif;
  font-weight: 400;
  font-size: 2.25rem;
  -webkit-letter-spacing: -0.03125rem;
  -moz-letter-spacing: -0.03125rem;
  -ms-letter-spacing: -0.03125rem;
  letter-spacing: -0.03125rem;
  line-height: 1.25;
  text-transform: capitalize;
  color: rgb(36,28,21);
  margin: 0;
  margin-bottom: 0.5rem;
  grid-column: center-start/center-end;
  word-break: break-word;
}

@media screen and (min-width:40em) {
  .emotion-2 {
    font-size: 2.875rem;
  }
}

.emotion-4 {
  font-family: Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif;
  font-weight: 400;
  font-size: 1.25rem;
  -webkit-letter-spacing: -0.0125rem;
  -moz-letter-spacing: -0.0125rem;
  -ms-letter-spacing: -0.0125rem;
  letter-spacing: -0.0125rem;
  line-height: 1.25;
  text-transform: capitalize;
  color: rgb(36,28,21);
  margin: 0;
  margin-bottom: 1rem;
  grid-column: center-start/center-end;
  word-break: break-word;
}

@media screen and (min-width:40em) {
  .emotion-4 {
    font-size: 1.75rem;
  }
}

.emotion-8 {
  grid-column: center-start/center-end;
}

.emotion-6 {
  font-family: Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  color: rgb(36,28,21);
  margin: 0;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  margin-right: 0.5rem;
  margin-bottom: 0.5rem;
  word-break: break-word;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-transition: all 0.15s linear;
  transition: all 0.15s linear;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  box-shadow: inset 0 0 0 0.0625rem rgba(36,28,21,0.4);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: transparent;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: middle;
}

.emotion-6:visited:hover,
.emotion-6:hover,
.emotion-6:focus {
  color: inherit;
}

.emotion-6:visited {
  color: inherit;
}

.emotion-6:hover,
.emotion-6:focus {
  color: rgb(64,59,59);
  box-shadow: inset 0 0 0 0.0625rem rgb(216,234,204);
  background-color: rgb(216,234,204);
}

<header
  class="emotion-10 emotion-9"
>
  <h1
    class="emotion-0 emotion-1"
    data-testid="name"
  >
    Andy Kenward
  </h1>
  <h2
    class="emotion-2 emotion-1"
    data-testid="role"
  >
    Senior Front-End Engineer
  </h2>
  <h3
    class="emotion-4 emotion-1"
    data-testid="update"
  >
    Hire Me
  </h3>
  <div
    class="emotion-8 emotion-9"
    data-testid="links"
  >
    <a
      class="emotion-6 emotion-1"
      data-testid="link"
      href="/link"
      rel="noopener noreferrer"
      target="_blank"
    >
      Github
    </a>
  </div>
</header>
`;

exports[`<Header /> should render when \`links\` is \`null\` 1`] = `
.emotion-8 {
  background-color: rgb(216,234,204);
  display: grid;
  grid-template-columns: 32px [center-start] 1fr [center-end] 32px;
  padding-top: 4rem;
  padding-bottom: 4rem;
}

@media screen and (min-width:40em) {
  .emotion-8 {
    grid-template-columns: minmax(32px,15%) [center-start] 1fr [center-end] minmax(32px,15%);
  }
}

@media screen and (min-width:40em) {
  .emotion-8 {
    padding-top: 8rem;
    padding-bottom: 8rem;
  }
}

.emotion-0 {
  font-family: "Cooper Light",Georgia,Times,"Times New Roman",serif;
  font-weight: 300;
  font-size: 3rem;
  line-height: 1.090909091;
  text-transform: capitalize;
  color: rgb(36,28,21);
  margin: 0;
  grid-column: center-start/center-end;
  word-break: break-word;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    font-size: 4rem;
  }
}

@media screen and (min-width:52em) {
  .emotion-0 {
    font-size: 5rem;
  }
}

.emotion-2 {
  font-family: Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif;
  font-weight: 400;
  font-size: 2.25rem;
  -webkit-letter-spacing: -0.03125rem;
  -moz-letter-spacing: -0.03125rem;
  -ms-letter-spacing: -0.03125rem;
  letter-spacing: -0.03125rem;
  line-height: 1.25;
  text-transform: capitalize;
  color: rgb(36,28,21);
  margin: 0;
  margin-bottom: 0.5rem;
  grid-column: center-start/center-end;
  word-break: break-word;
}

@media screen and (min-width:40em) {
  .emotion-2 {
    font-size: 2.875rem;
  }
}

.emotion-4 {
  font-family: Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif;
  font-weight: 400;
  font-size: 1.25rem;
  -webkit-letter-spacing: -0.0125rem;
  -moz-letter-spacing: -0.0125rem;
  -ms-letter-spacing: -0.0125rem;
  letter-spacing: -0.0125rem;
  line-height: 1.25;
  text-transform: capitalize;
  color: rgb(36,28,21);
  margin: 0;
  margin-bottom: 1rem;
  grid-column: center-start/center-end;
  word-break: break-word;
}

@media screen and (min-width:40em) {
  .emotion-4 {
    font-size: 1.75rem;
  }
}

<header
  class="emotion-8 emotion-9"
>
  <h1
    class="emotion-0 emotion-1"
    data-testid="name"
  >
    Andy Kenward
  </h1>
  <h2
    class="emotion-2 emotion-1"
    data-testid="role"
  >
    Senior Front-End Engineer
  </h2>
  <h3
    class="emotion-4 emotion-1"
    data-testid="update"
  >
    Hire Me
  </h3>
  <h4
    class="emotion-4 emotion-1"
    data-testid="available"
  >
    Available from 
    April 2020
  </h4>
</header>
`;

exports[`<Header /> should render when \`name\` is \`null\` 1`] = `
.emotion-10 {
  background-color: rgb(216,234,204);
  display: grid;
  grid-template-columns: 32px [center-start] 1fr [center-end] 32px;
  padding-top: 4rem;
  padding-bottom: 4rem;
}

@media screen and (min-width:40em) {
  .emotion-10 {
    grid-template-columns: minmax(32px,15%) [center-start] 1fr [center-end] minmax(32px,15%);
  }
}

@media screen and (min-width:40em) {
  .emotion-10 {
    padding-top: 8rem;
    padding-bottom: 8rem;
  }
}

.emotion-0 {
  font-family: Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif;
  font-weight: 400;
  font-size: 2.25rem;
  -webkit-letter-spacing: -0.03125rem;
  -moz-letter-spacing: -0.03125rem;
  -ms-letter-spacing: -0.03125rem;
  letter-spacing: -0.03125rem;
  line-height: 1.25;
  text-transform: capitalize;
  color: rgb(36,28,21);
  margin: 0;
  margin-bottom: 0.5rem;
  grid-column: center-start/center-end;
  word-break: break-word;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    font-size: 2.875rem;
  }
}

.emotion-2 {
  font-family: Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif;
  font-weight: 400;
  font-size: 1.25rem;
  -webkit-letter-spacing: -0.0125rem;
  -moz-letter-spacing: -0.0125rem;
  -ms-letter-spacing: -0.0125rem;
  letter-spacing: -0.0125rem;
  line-height: 1.25;
  text-transform: capitalize;
  color: rgb(36,28,21);
  margin: 0;
  margin-bottom: 1rem;
  grid-column: center-start/center-end;
  word-break: break-word;
}

@media screen and (min-width:40em) {
  .emotion-2 {
    font-size: 1.75rem;
  }
}

.emotion-8 {
  grid-column: center-start/center-end;
}

.emotion-6 {
  font-family: Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  color: rgb(36,28,21);
  margin: 0;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  margin-right: 0.5rem;
  margin-bottom: 0.5rem;
  word-break: break-word;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-transition: all 0.15s linear;
  transition: all 0.15s linear;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  box-shadow: inset 0 0 0 0.0625rem rgba(36,28,21,0.4);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: transparent;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: middle;
}

.emotion-6:visited:hover,
.emotion-6:hover,
.emotion-6:focus {
  color: inherit;
}

.emotion-6:visited {
  color: inherit;
}

.emotion-6:hover,
.emotion-6:focus {
  color: rgb(64,59,59);
  box-shadow: inset 0 0 0 0.0625rem rgb(216,234,204);
  background-color: rgb(216,234,204);
}

<header
  class="emotion-10 emotion-9"
>
  <h2
    class="emotion-0 emotion-1"
    data-testid="role"
  >
    Senior Front-End Engineer
  </h2>
  <h3
    class="emotion-2 emotion-1"
    data-testid="update"
  >
    Hire Me
  </h3>
  <h4
    class="emotion-2 emotion-1"
    data-testid="available"
  >
    Available from 
    April 2020
  </h4>
  <div
    class="emotion-8 emotion-9"
    data-testid="links"
  >
    <a
      class="emotion-6 emotion-1"
      data-testid="link"
      href="/link"
      rel="noopener noreferrer"
      target="_blank"
    >
      Github
    </a>
  </div>
</header>
`;

exports[`<Header /> should render when \`role\` is \`null\` 1`] = `
.emotion-10 {
  background-color: rgb(216,234,204);
  display: grid;
  grid-template-columns: 32px [center-start] 1fr [center-end] 32px;
  padding-top: 4rem;
  padding-bottom: 4rem;
}

@media screen and (min-width:40em) {
  .emotion-10 {
    grid-template-columns: minmax(32px,15%) [center-start] 1fr [center-end] minmax(32px,15%);
  }
}

@media screen and (min-width:40em) {
  .emotion-10 {
    padding-top: 8rem;
    padding-bottom: 8rem;
  }
}

.emotion-0 {
  font-family: "Cooper Light",Georgia,Times,"Times New Roman",serif;
  font-weight: 300;
  font-size: 3rem;
  line-height: 1.090909091;
  text-transform: capitalize;
  color: rgb(36,28,21);
  margin: 0;
  grid-column: center-start/center-end;
  word-break: break-word;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    font-size: 4rem;
  }
}

@media screen and (min-width:52em) {
  .emotion-0 {
    font-size: 5rem;
  }
}

.emotion-2 {
  font-family: Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif;
  font-weight: 400;
  font-size: 1.25rem;
  -webkit-letter-spacing: -0.0125rem;
  -moz-letter-spacing: -0.0125rem;
  -ms-letter-spacing: -0.0125rem;
  letter-spacing: -0.0125rem;
  line-height: 1.25;
  text-transform: capitalize;
  color: rgb(36,28,21);
  margin: 0;
  margin-bottom: 1rem;
  grid-column: center-start/center-end;
  word-break: break-word;
}

@media screen and (min-width:40em) {
  .emotion-2 {
    font-size: 1.75rem;
  }
}

.emotion-8 {
  grid-column: center-start/center-end;
}

.emotion-6 {
  font-family: Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  color: rgb(36,28,21);
  margin: 0;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  margin-right: 0.5rem;
  margin-bottom: 0.5rem;
  word-break: break-word;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-transition: all 0.15s linear;
  transition: all 0.15s linear;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  box-shadow: inset 0 0 0 0.0625rem rgba(36,28,21,0.4);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: transparent;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: middle;
}

.emotion-6:visited:hover,
.emotion-6:hover,
.emotion-6:focus {
  color: inherit;
}

.emotion-6:visited {
  color: inherit;
}

.emotion-6:hover,
.emotion-6:focus {
  color: rgb(64,59,59);
  box-shadow: inset 0 0 0 0.0625rem rgb(216,234,204);
  background-color: rgb(216,234,204);
}

<header
  class="emotion-10 emotion-9"
>
  <h1
    class="emotion-0 emotion-1"
    data-testid="name"
  >
    Andy Kenward
  </h1>
  <h3
    class="emotion-2 emotion-1"
    data-testid="update"
  >
    Hire Me
  </h3>
  <h4
    class="emotion-2 emotion-1"
    data-testid="available"
  >
    Available from 
    April 2020
  </h4>
  <div
    class="emotion-8 emotion-9"
    data-testid="links"
  >
    <a
      class="emotion-6 emotion-1"
      data-testid="link"
      href="/link"
      rel="noopener noreferrer"
      target="_blank"
    >
      Github
    </a>
  </div>
</header>
`;

exports[`<Header /> should render when \`update\` is \`null\` 1`] = `
.emotion-10 {
  background-color: rgb(216,234,204);
  display: grid;
  grid-template-columns: 32px [center-start] 1fr [center-end] 32px;
  padding-top: 4rem;
  padding-bottom: 4rem;
}

@media screen and (min-width:40em) {
  .emotion-10 {
    grid-template-columns: minmax(32px,15%) [center-start] 1fr [center-end] minmax(32px,15%);
  }
}

@media screen and (min-width:40em) {
  .emotion-10 {
    padding-top: 8rem;
    padding-bottom: 8rem;
  }
}

.emotion-0 {
  font-family: "Cooper Light",Georgia,Times,"Times New Roman",serif;
  font-weight: 300;
  font-size: 3rem;
  line-height: 1.090909091;
  text-transform: capitalize;
  color: rgb(36,28,21);
  margin: 0;
  grid-column: center-start/center-end;
  word-break: break-word;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    font-size: 4rem;
  }
}

@media screen and (min-width:52em) {
  .emotion-0 {
    font-size: 5rem;
  }
}

.emotion-2 {
  font-family: Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif;
  font-weight: 400;
  font-size: 2.25rem;
  -webkit-letter-spacing: -0.03125rem;
  -moz-letter-spacing: -0.03125rem;
  -ms-letter-spacing: -0.03125rem;
  letter-spacing: -0.03125rem;
  line-height: 1.25;
  text-transform: capitalize;
  color: rgb(36,28,21);
  margin: 0;
  margin-bottom: 0.5rem;
  grid-column: center-start/center-end;
  word-break: break-word;
}

@media screen and (min-width:40em) {
  .emotion-2 {
    font-size: 2.875rem;
  }
}

.emotion-4 {
  font-family: Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif;
  font-weight: 400;
  font-size: 1.25rem;
  -webkit-letter-spacing: -0.0125rem;
  -moz-letter-spacing: -0.0125rem;
  -ms-letter-spacing: -0.0125rem;
  letter-spacing: -0.0125rem;
  line-height: 1.25;
  text-transform: capitalize;
  color: rgb(36,28,21);
  margin: 0;
  margin-bottom: 1rem;
  grid-column: center-start/center-end;
  word-break: break-word;
}

@media screen and (min-width:40em) {
  .emotion-4 {
    font-size: 1.75rem;
  }
}

.emotion-8 {
  grid-column: center-start/center-end;
}

.emotion-6 {
  font-family: Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  color: rgb(36,28,21);
  margin: 0;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  margin-right: 0.5rem;
  margin-bottom: 0.5rem;
  word-break: break-word;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-transition: all 0.15s linear;
  transition: all 0.15s linear;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  box-shadow: inset 0 0 0 0.0625rem rgba(36,28,21,0.4);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: transparent;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: middle;
}

.emotion-6:visited:hover,
.emotion-6:hover,
.emotion-6:focus {
  color: inherit;
}

.emotion-6:visited {
  color: inherit;
}

.emotion-6:hover,
.emotion-6:focus {
  color: rgb(64,59,59);
  box-shadow: inset 0 0 0 0.0625rem rgb(216,234,204);
  background-color: rgb(216,234,204);
}

<header
  class="emotion-10 emotion-9"
>
  <h1
    class="emotion-0 emotion-1"
    data-testid="name"
  >
    Andy Kenward
  </h1>
  <h2
    class="emotion-2 emotion-1"
    data-testid="role"
  >
    Senior Front-End Engineer
  </h2>
  <h4
    class="emotion-4 emotion-1"
    data-testid="available"
  >
    Available from 
    April 2020
  </h4>
  <div
    class="emotion-8 emotion-9"
    data-testid="links"
  >
    <a
      class="emotion-6 emotion-1"
      data-testid="link"
      href="/link"
      rel="noopener noreferrer"
      target="_blank"
    >
      Github
    </a>
  </div>
</header>
`;
